app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -224,10 +224,15 @@ public void onValueChanged(@NonNull Boolean loading) {
                     progressView.setProgress(5);
                     progressView.setVisibility(View.VISIBLE);
                 } else {
-                    backgroundTransitionGroup.startTransition(ANIMATION_DURATION);
 
-                    progressView.setProgress(progressView.getMax());
-                    progressView.setVisibility(View.GONE);
+                    if (progressView.getVisibility() == View.VISIBLE) {
```

Comment:
I think we could go one level higher and apply this patch to SessionCallbackProxy:
https://github.com/mozilla-mobile/focus-android/blob/d94251c4d7c262f26bb8ba22e5a218cff2e7354f/app/src/main/java/org/mozilla/focus/session/SessionCallbackProxy.java#L40

As it looks like onPageFinished() gets called multiple times by the WebView here. We should only set `session.setLoading(false);` if we are loading (`session.getLoading().getValue()`). We should still always do the setSecure() call in the next line though (in case it changes).

Reply:
Well thank you, I will come back with a commit very soon.